### PR TITLE
HDFS-14761. RBF: MountTableResolver cannot invalidate cache correctly

### PR DESCRIPTION
**Pull request by Cosss7**

### CODE REVIEW

**Comment by hadoop-yetus** [comment]
:broken_heart: **-1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| 0 | reexec | 42 | Docker mode activated. |
||| _ Prechecks _ |
| +1 | dupname | 0 | No case conflicting files found. |
| +1 | @author | 0 | The patch does not contain any @author tags. |
| +1 | test4tests | 0 | The patch appears to include 1 new or modified test files. |
||| _ trunk Compile Tests _ |
| +1 | mvninstall | 1065 | trunk passed |
| +1 | compile | 27 | trunk passed |
| +1 | checkstyle | 30 | trunk passed |
| +1 | mvnsite | 34 | trunk passed |
| +1 | shadedclient | 689 | branch has no errors when building and testing our client artifacts. |
| +1 | javadoc | 32 | trunk passed |
| 0 | spotbugs | 62 | Used deprecated FindBugs config; considering switching to SpotBugs. |
| +1 | findbugs | 59 | trunk passed |
||| _ Patch Compile Tests _ |
| +1 | mvninstall | 26 | the patch passed |
| +1 | compile | 22 | the patch passed |
| +1 | javac | 22 | the patch passed |
| +1 | checkstyle | 13 | the patch passed |
| +1 | mvnsite | 25 | the patch passed |
| +1 | whitespace | 0 | The patch has no whitespace issues. |
| +1 | shadedclient | 697 | patch has no errors when building and testing our client artifacts. |
| +1 | javadoc | 32 | the patch passed |
| +1 | findbugs | 71 | the patch passed |
||| _ Other Tests _ |
| -1 | unit | 1353 | hadoop-hdfs-rbf in the patch failed. |
| +1 | asflicense | 25 | The patch does not generate ASF License warnings. |
| | | 4352 | |


| Reason | Tests |
|-------:|:------|
| Failed junit tests | hadoop.hdfs.server.federation.router.TestRouterWithSecureStartup |
|   | hadoop.hdfs.server.federation.security.TestRouterHttpDelegationToken |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | Client=19.03.1 Server=19.03.1 base: https://builds.apache.org/job/hadoop-multibranch/job/PR-1334/1/artifact/out/Dockerfile |
| GITHUB PR | https://github.com/apache/hadoop/pull/1334 |
| Optional Tests | dupname asflicense compile javac javadoc mvninstall mvnsite unit shadedclient findbugs checkstyle |
| uname | Linux 2ac5d110f440 4.4.0-138-generic #164-Ubuntu SMP Tue Oct 2 17:16:02 UTC 2018 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | personality/hadoop.sh |
| git revision | trunk / ee7c261 |
| Default Java | 1.8.0_212 |
| unit | https://builds.apache.org/job/hadoop-multibranch/job/PR-1334/1/artifact/out/patch-unit-hadoop-hdfs-project_hadoop-hdfs-rbf.txt |
|  Test Results | https://builds.apache.org/job/hadoop-multibranch/job/PR-1334/1/testReport/ |
| Max. process+thread count | 1604 (vs. ulimit of 5500) |
| modules | C: hadoop-hdfs-project/hadoop-hdfs-rbf U: hadoop-hdfs-project/hadoop-hdfs-rbf |
| Console output | https://builds.apache.org/job/hadoop-multibranch/job/PR-1334/1/console |
| versions | git=2.7.4 maven=3.3.9 findbugs=3.1.0-RC1 |
| Powered by | Apache Yetus 0.10.0 http://yetus.apache.org |


This message was automatically generated.

File: hadoop-hdfs-project/hadoop-hdfs-rbf/src/test/java/org/apache/hadoop/hdfs/server/federation/resolver/TestMountTableResolver.java [hunk 1]
```diff
@@ -704,4 +704,26 @@ public void testLocationCache() throws Exception {
     mountTable.removeEntry("/testlocationcache");
     mountTable.removeEntry("/anothertestlocationcache");
   }
+
+  @Test
+  public void testInvalidateCache() throws Exception {
```
Comment (goiri) [inline hadoop-hdfs-project/hadoop-hdfs-rbf/src/test/java/org/apache/hadoop/hdfs/server/federation/resolver/TestMountTableResolver.java]: Add a comment describing what the case to test is about.

Comment (Cosss7) [reply hadoop-hdfs-project/hadoop-hdfs-rbf/src/test/java/org/apache/hadoop/hdfs/server/federation/resolver/TestMountTableResolver.java]: Thanks for your comments. I have added some comments. Pending jenkins. Please review again.

Comment (goiri) [reply hadoop-hdfs-project/hadoop-hdfs-rbf/src/test/java/org/apache/hadoop/hdfs/server/federation/resolver/TestMountTableResolver.java]: The high level description should go as a javadoc.
You can add the particular comments inline.

Comment (Cosss7) [reply hadoop-hdfs-project/hadoop-hdfs-rbf/src/test/java/org/apache/hadoop/hdfs/server/federation/resolver/TestMountTableResolver.java]: Thanks again. New commit here. Pending jenkins.

**Pull request merged**